core/chains/evm/client/simulated_backend_client.go

```diff
@@ -64,37 +73,82 @@ func (c *SimulatedBackendClient) checkEthCallArgs(
 			"must be an eth.CallArgs, got %+#v", args[0])
 	}
 	blockNumber, err := c.blockNumber(args[1])
-	if err != nil || blockNumber.Cmp(c.currentBlockNumber()) != 0 {
+	if err != nil {
 		return nil, nil, fmt.Errorf("fourth arg to SimulatedBackendClient.Call "+
-			"must be the string \"latest\", or a *big.Int equal to current "+
-			"blocknumber, got %#+v", args[1])
+			"must be the string \"latest\", or a *big.Int, got %#+v", args[1])
+	}
+
+	toAddr, err := interfaceToAddress(callArgs["to"])
+	if err != nil {
+		return nil, nil, err
 	}
+
+	// to and from need to map to a common.Address but could come in as a string
 	ca := CallArgs{
-		From: callArgs["from"].(common.Address),
-		To:   *callArgs["to"].(*common.Address),
+		To:   toAddr,
 		Data: callArgs["data"].(hexutil.Bytes),
 	}
+
+	// from is optional in the standard client; default to 0x when missing
+	if value, ok := callArgs["from"]; ok {
+		addr, err := interfaceToAddress(value)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		ca.From = addr
+	} else {
+		ca.From = common.HexToAddress("0x")
```

Comment:
This seems weird. Why don't we just err here?
FromAddress being 0x will anyways not work i later part of code.

Reply:
I have it like this because other parts of the code don't provide a 'From' value as I suppose the production client doesn't require it. I tried to model the simulated client to follow the same behavior.